src/toil/lib/aws/__init__.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (C) 2015-2021 Regents of the University of California
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Caches all boto3 clients we instantiate."""
+from functools import lru_cache
+
+
+@lru_cache(maxsize=None)
+def client(*args, **kwargs):
+    import boto3
+    return boto3.client(*args, **kwargs)
```

Comment:
I'm going to need to glue this onto the logic I've been using to establish a properly credential-cached Boto3 session for the AWS provisioner work I've been doing.

Reply:
That sounds good to me.

Reply:
I feel like this is going to create a default session to get this client, while we now have https://github.com/DataBiosphere/toil/blob/65699cd653d67e9717fdd9a35946550ea3ea21db/src/toil/lib/ec2.py#L74-L80 which is supposed to be the one place to get sessions.

We also might want to think about how/whether things should be split between `toil.lib.ec2` and `toil.lib.aws`.